common/djangoapps/student/views.py

```diff
@@ -1718,7 +1722,7 @@ def create_account_with_params(request, params):
     # Perform operations within a transaction that are critical to account creation
     with transaction.atomic():
         # first, create the account
-        (user, profile, registration) = _do_create_account(form, custom_form)
+        (user, profile, registration) = _do_create_account(form, custom_form, site=getattr(request, 'site', None))
```

Comment:
We should always have the [site middleware](https://docs.djangoproject.com/en/1.10/ref/contrib/sites/#site-middleware) enabled. Use `request.site` and avoid the headaches that will crop up when there is no site.